### DISCUSSION TITLE
How can i retain 7 days worth of pcap

### DISCUSSION THREAD

**Discussion by thegreekman** [Security-Onion-Solutions/securityonion#5313] [Unsupported Versions]
Hello!  I am on the latest version of security onion. Currently pcap generates about ~1TB of data per day. Is there anyway to reduce this without deleting the pcap files? I added the bpf to remove port 443 traffic, but there is still a lot of traffic. 

Any help is much appreciated

**Comment by facyber** [comment]
You can use Stenographer in combination with mounted shared backup folder where you would store extracted PCAP files.

You can create a BASH script and set a cronjob to create a backup of PCAP files every hour for example. Command that you would use for this is `docker exec -it so-steno stenoread "after 2h ago and before 1h ago" -w /tmp/new.pcap`. This command will extract PCAP into the _/nsm/pcaptmp/_ folder under name **new.pcap**. You need to run this command 

The script process flow would look something like this:
1. Script run docker exec command and creates a _new.pcap_ file into the _/nsm/pcaptmp/_
2. Rename the file to something like _new-{date:time}.pcap_ so you don't get file overwritten
3. You move (not copy) the content of pcaptmp folder to the mounted folder (_/mnt/shared-folder_ for example or whatever you call it).

More information about Stenographer and reading pcap files can be found in [SO documentation](https://docs.securityonion.net/en/2.3/stenographer.html?highlight=so-steno#analysis), and more about stenographer query format can be found [here](https://github.com/google/stenographer#querying).

There is also a [dump script for Stenographer ](https://github.com/aeppert/shell_scripts/blob/master/dump_steno.sh)but I haven't tested it. Maybe you can use it as a base for your script.

**Reply by thegreekman** [reply]
I see, thanks. And as for reducing the size of pcap files do you have any reccomendations?

**Reply by facyber** [reply]
Not that I am aware of, except to backuo in smaller time ibtervals, half an hour of data for example.

**Comment by PatRussell87** [comment]
I had a similar issue and just had to review the traffic I was monitoring and start filtering out what I wasn't interested in. Most of this ended up being stuff like management traffic between storage arrays. Very specific endpoints and ports made it easy to filter out. I was able to reduce my PCAP storage to the point where I have several weeks available rather than a few hours.

**Reply by thegreekman** [reply]
Was your filtering included for zeek and suricata or just steno?

**Comment by PatRussell87** [comment]
I use different filtering for each of those. For some traffic, I may not want to alert on it but still want to capture and have the PCAP. There is a little overlap in the filtering between tools, but not much.

**Reply by thegreekman** [reply]
May I ask, what is your average count of logs  per day?

**Comment by PatRussell87** [comment]
Looking at my Kibana dashboard, 2.5 million per day. Right now, that is from one sensor. Adding several more once I get some things ironed out and hardware comes in.

**Reply by thegreekman** [reply]
I see. Currently generating around 25-30 million logs a day on one node(Standalone deployment). Over 50% of those logs are from syslog and zeek. Do you know if there is a way to determine what traffic is taking up the most space in the pcap files? I've blocked https, snmb, and ntp traffic so far which improved my retainment from several hours to almost 24 hours, but that is still not close to 7 days haha.

**Reply by PatRussell87** [reply]
I started looking at the traffic in both Kibana and PCAPs and filtering it down from there. My deployment is distributed so that will be different as far as storage locations. All I needed to deal with was the actual PCAPs on my sensor node since Zeek and Suricata info is kept in the storage node.

I think using the Discover tool and dashboards in Kibana will help a lot. Work from the dashboards based on highest traffic captures and start paring it down.

**Reply by thegreekman** [reply]
Ok thanks. How do you look at the traffic in the pcap files?

**Comment by PatRussell87** [comment]
From the PCAP tool in SO you can specify time windows to recall a capture. Click the "+" in the upper left corner to open the query window. Once the file is generated, you can view traffic info directly in the PCAP web interface.
